File: pkg/worker/event.go
package worker

import "encoding/json"

type Event struct {
	Provider   string                 `json:"provider"`
	Type       string                 `json:"type"`
	Topic      string                 `json:"topic"`
	Metadata   map[string]string      `json:"metadata"`
	Payload    json.RawMessage        `json:"payload"`
	Normalized map[string]interface{} `json:"normalized"`
	Client     interface{}            `json:"-"`
}

File: sdk/typescript/worker/src/event.ts
import { Buffer } from "node:buffer";

export interface Event {
  provider: string;
  type: string;
  topic: string;
  metadata: Record<string, string>;
  payload: Buffer;
  normalized?: Record<string, unknown>;
  requestId?: string;
  installationId?: string;
  logId?: string;
  client?: unknown;
}
